Update some libraries, fixed asm version numbers throughout, fixed xml issue with escapes

hazendaz opened the pull request:
----

Make sure these boxes are checked before submitting your PR -- thank you!

- [X ] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

hazendaz commented:
This started out as an effort of seeing commons-lang still used in the spot bugs maven plugin coming from here.  Also comes from maven too, but felt the need to come over here and fix it.  Through that process, I found a number of other things in same area of code and had to spend a good deal of time looking at that encoding issue.  If 'html' was used for encoding that does work but clearly it is xml.  Further commons-lang2 is really old.  I couldn't find anything that definatively said it was needed to escape in xml, just html.

Other notes, github actions / travis both fail when coming from a fork.  While not a huge concern here, probably something that should be looked at to resolve so forks are not trying to do all the same things.

KengoTODA commented on spotbugs/src/test/java/edu/umd/cs/findbugs/util/StringsTest.java:
```diff
@@ -21,13 +21,14 @@
 
     public static String[] escapedStrings = {
         // mixed entities/unicode escape sequences
-        "a b c 1 2 3 &amp; &lt; &gt; &quot; &apos; \\u0005 \\u0013 &#955; \\\\u0007",
```
is it necessary change? 🤔

hazendaz commented:
Yes.  The escape value doesn't apply to XML.  The character is valid.  Greek lambda.  It only applies to HTML.  You can trust by flipping time HTML encode which will show the difference.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________
From: Kengo TODA <notifications@github.com>
Sent: Sunday, March 29, 2020 6:31:53 AM
To: spotbugs/spotbugs <spotbugs@noreply.github.com>
Cc: Jeremy Landis <jeremylandis@hotmail.com>; Author <author@noreply.github.com>
Subject: Re: [spotbugs/spotbugs] Update some libraries, fixed asm version numbers throughout, fixed xml issue with escapes (#1111)


@KengoTODA commented on this pull request.

________________________________

In spotbugs/src/test/java/edu/umd/cs/findbugs/util/StringsTest.java<https://github.com/spotbugs/spotbugs/pull/1111#discussion_r399777500>:

>  import org.junit.Test;



 public class StringsTest {



     public static String[] escapedStrings = {

         // mixed entities/unicode escape sequences

-        "a b c 1 2 3 &amp; &lt; &gt; &quot; &apos; \\u0005 \\u0013 &#955; \\\\u0007",



is it necessary change? 🤔

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/spotbugs/spotbugs/pull/1111#pullrequestreview-383395543>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAHODIZBSBP56TYIESMLWWTRJ4PRTANCNFSM4LV2EQZA>.
